### PR TITLE
Add premiumType

### PR DESCRIPTION
**Pull request by HeadTriXz**
premiumType can be used to check what type of nitro a user has. It can either return 0, 1 or 2.

> 0: None
> 1: Nitro Classic
> 2: Nitro

### CODE REVIEW

**Review by abalabahaha — DISMISSED** [review]

File: lib/structures/User.js [hunk 1]
```diff
@@ -41,6 +42,9 @@ class User extends Base {
         if(data.discriminator !== undefined) {
             this.discriminator = data.discriminator;
         }
+        if(data.premium_type !== undefined) {
+            this.premiumType = data.premium_type;
```
Comment (bsian03) [inline lib/structures/User.js:46]: Can you also add this to the `toJSON` method at the bottom of the file

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: lib/structures/User.js (hunk 1 above)
Comment (HeadTriXz) [reply lib/structures/User.js:46]: Done!

File: lib/structures/User.js [hunk 2]
```diff
@@ -14,7 +14,8 @@ const Endpoints = require("../rest/Endpoints");
 * @prop {String} discriminator The discriminator of the user
 * @prop {String} id The ID of the user
 * @prop {String} mention A string that mentions the user
-* @prop {Number?} publicFlags Publicly visible flags for this user
+* @prop {Number?} premiumType The type of nitro the user has
+* @prop {Number?} publicFlags Publicly vicible flags for this user
```
Comment (bsian03) [inline lib/structures/User.js]: Fix the typo here
```suggestion
* @prop {Number?} publicFlags Publicly visible flags for this user
```

**Review by bsian03 — CHANGES_REQUESTED** [review]
Just so you know, all PRs should be made with the dev branch to the dev branch -  the typo was actually fixed in dev in a previous PR

**Comment by HeadTriXz** [comment]
> Just so you know, all PRs should be made with the dev branch to the dev branch - the typo was actually fixed in dev in a previous PR

Yeah, I noticed. My bad. Would you like me to create a new PR using the dev branch or is it fine like this?

**Comment by bsian03** [comment]
It's fine here, just mainly to prevent awkward conflicts from happening if you're not on the latest commit

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**